// Copyright (c) 2023, Cisco Systems, Inc. and/or its affiliates.
// All rights reserved.
// See LICENSE file in this distribution.
// SPDX-License-Identifier: Apache-2.0

syntax = "proto3";

option java_package = "com.cisco.tiedie.proto";
option java_multiple_files = true;

package tiedie;

message DataSubscription {
    int64 timestamp = 1;
    string ap_macaddr = 2;
    optional string device_id = 3;
    bytes data = 4;

    oneof subscription {
        BLESubscription ble_subscription = 5;
        BLEAdvertisement ble_advertisement = 6;
        ZigbeeSubscription zigbee = 7;
        RawPayload raw_payload = 8;
        BLEConnectionStatus ble_connection_status = 9;
    }

    message BLESubscription {
        optional string service_uuid = 1;
        optional string characteristic_uuid = 2;
    }

    message BLEAdvertisement {
        string mac_address = 1;
        optional int32 rssi = 2;
    }
    
    message ZigbeeSubscription {
        optional int32 endpoint_id = 1;
        optional int32 cluster_id = 2;
        optional int32 attribute_id = 3;
        optional int32 attribute_type = 4;
    }

    message BLEConnectionStatus {
        string mac_address = 1;
        bool connected = 2;
        optional int32 reason = 3;
    }
    
    message RawPayload {
        optional string context_id = 1;
    }
}